src/application/CreateProjectModal.tsx

```diff
@@ -0,0 +1,65 @@
+import { useAtomValue, useSetAtom } from 'jotai';
+import { FormEventHandler, MouseEventHandler, useState } from 'react';
+
+import { createProjectModalAtoms } from '../atoms/projectState';
+import { Button } from '../components/Button';
+import { autoFocus } from '../lib/autoFocusAndSelect';
+import { cx } from '../lib/cx';
+
+export function CreateProjectModal() {
+  const [name, setName] = useState('');
+
+  const isVisible = useAtomValue(createProjectModalAtoms.visibleAtom);
+  const closeModal = useSetAtom(createProjectModalAtoms.closeAtom);
+
+  const cleanupName = (_name: string) => _name.replace(/[^a-zA-Z0-9-_ ]/g, '');
+
+  const handleFormSubmit: FormEventHandler = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (!name) {
+      return;
+    }
+    closeModal(name.trim());
+  };
+  const handleOnCancel: MouseEventHandler<HTMLButtonElement> = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    closeModal(null);
+  };
+
+  if (!isVisible) {
+    return <></>;
+  }
+
+  return (
+    <div className="fixed left-0 top-0 z-modals flex h-screen w-screen items-center justify-center">
+      <div className="pointer-events-none absolute left-0 top-0 h-full w-full bg-slate-400/60" />
```

Comment:
A generic Modal component will be added in #479. Do you want to wait for it before merging this PR?

Reply:
I will.